### PR TITLE
[FIX]: BBI residuals vs Sampling example

### PR DESCRIPTION
**Pull request by achamma723**
This PR implements the ```Residuals``` vs ```Sampling``` example using BBI.

### CODE REVIEW

**Comment by bthirion** [comment]
I guess we need  #3 to be merged before looking at that one.

**Comment by achamma723** [comment]
@bthirion talking about this one we might need more cpu cores to run the experiments otherwise we will just show a simple scaled example

**Comment by bthirion** [comment]
Maybe just use a single scaled example for the time being.

File: examples/plot_diabetes_variable_importance_example.py [hunk 1]
```diff
@@ -2,8 +2,44 @@
...
+permutation breaks the relationship between the variable of interest and the
+outcome. Following, the loss score is checked before and after this
+substitution for any significant drop in the performance which reflects the
+significance of this variable to predict the outcome. This ease-to-use solution
+is demonstrated, in the work by
+:footcite:t:`stroblConditionalVariableImportance2008`, to be affected by the
+degree of correlation between the variables, thus biased towards truly
+non-significant variables highly correlated with the significant ones and
+creating fake significant variables. They introduced a solution for the Random
+Forest estimator based on the conditional sampling by performing sub-groups
```
Comment (bthirion) [inline examples/plot_diabetes_variable_importance_example.py:18]:
```suggestion
Forest estimator based on conditional sampling by performing sub-groups
```

File: examples/plot_residuals_sampling.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+dict_hyper = {"max_depth": [2, 5, 10, 20]}
+
+#############################################################################
+# Generate the synthetic data
+# ---------------------------
+# The function below generates the correlation matrix between the variables
+# according to the provided degrees of correlation (intra + inter). (inter_cor)
+# indicates the degree of correlation between the variables/groups whereas
+# (intra_cor) specifies the corresponding degree between the variables within
+# each group. For the single-level case, the (n_blocks) is set to 1 and the
```
Comment (bthirion) [inline examples/plot_residuals_sampling.py:38]:
```suggestion
# each group. For the single-level case, `n_blocks` is set to 1 and the
```

File: examples/plot_residuals_sampling.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+# (intra_cor) illustrates the correlation between the variables.
```
Comment (bthirion) [inline examples/plot_residuals_sampling.py:39]:
```suggestion
# `intra_cor` is the unique correlation between variables.
```

File: examples/plot_residuals_sampling.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    y = prod_signal + sigma_noise * rng.normal(size=prod_signal.shape[0])
+
+    return data, y
+
+
+#############################################################################
+# Processing across multiple permutations
+# ---------------------------------------
+# In order to get statistical significance with p-values, we run the experiments
+# across 100 repetitions.
```
Comment (bthirion) [inline examples/plot_residuals_sampling.py:101]: 50 ?

**Review by bthirion — COMMENTED** [review]
Mostly LGTM, But I can't see the generated doc...

**Comment by achamma723** [comment]
@bthirion, this PR is ready for another review

File: examples/plot_diabetes_variable_importance_example.py [hunk 5]
```diff
@@ -2,8 +2,44 @@
 Variable Importance on diabetes dataset
 =======================================
 
-This example compares the standard permutation approach for variable importance
-and its conditional variant on the diabetes dataset for the single-level case.
+Variable Importance estimates the influence of a given input variable to the
+prediction made by a model. To perform variable importance in a prediction
```
Comment (bthirion) [inline examples/plot_diabetes_variable_importance_example.py:6]:
```suggestion
prediction made by a model. To assess variable importance in a prediction
```

File: examples/plot_diabetes_variable_importance_example.py [hunk 6]
```diff
@@ -2,8 +2,44 @@
...
+buiding process. However, this solution is exclusive to the Random Forest and is
+costly with high-dimensional settings.
+:footcite:t:`Chamma_NeurIPS2023` introduced a new model-agnostic solution to
+bypass the limitations of the permutation approach under the use of the
+conditional schemes. The variable of interest does contain two types of
+information: 1) the relationship with the remaining variables and 2) the
+relationship with the outcome. The standard permutation, while breaking the
+relationship with the outcome, is also destroying the dependency with the
+remaining variables. Therefore, instead of directly permuting the variable of
+interest, the variable of interest is predicted by the mean of the remaining
```
Comment (bthirion) [inline examples/plot_diabetes_variable_importance_example.py:29]:
```suggestion
interest, the variable of interest is predicted by the remaining
```

**Review by bthirion — COMMENTED** [review]
Looks good overall. Can you fix the two minor  glitches ?

**Comment by achamma723** [comment]
@bthirion, this PR is ready to review

**Comment by bthirion** [comment]
LGTM, will merge now.

**Pull request merged**